arch/arm/src/arm/arm_undefinedinsn.c

```diff
@@ -42,7 +42,16 @@
 
 void arm_undefinedinsn(uint32_t *regs)
 {
-  _alert("Undefined instruction at 0x%" PRIx32 "\n", regs[REG_PC]);
+  if (regs[REG_PC] >= (uint32_t)_stext && regs[REG_PC] < (uint32_t)_etext)
+    {
+      _alert("Undefined instruction at 0x%" PRIx32 ":0x%" PRIx32 "\n",
```

Comment:
This might look like a segment:offset address like back in the DOS / PC days. Maybe add a space, like so:

```suggestion
      _alert("Undefined instruction at 0x%" PRIx32 ": 0x%" PRIx32 "\n",
```